tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/S3SingleSaveBlobStoreDAO.java

```diff
@@ -0,0 +1,37 @@
+package com.linagora.tmail.blob.blobid.list;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.blob.api.ObjectStoreException;
+import org.apache.james.blob.objectstorage.aws.S3BlobStoreConfiguration;
+import org.apache.james.blob.objectstorage.aws.S3BlobStoreDAO;
+import reactor.core.publisher.Mono;
+
+import javax.inject.Inject;
+import java.util.Objects;
+
+public class S3SingleSaveBlobStoreDAO extends S3BlobStoreDAO {
+    private final BlobIdList blobIdList;
+
+    @Inject
+    public S3SingleSaveBlobStoreDAO(S3BlobStoreConfiguration configuration, BlobIdList blobIdList) {
+        super(configuration);
+        this.blobIdList = blobIdList;
+    }
+
+    @Override
+    public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
+        return Mono.from(blobIdList.isStored(blobId))
```

Comment:
Maybe, I have a problem with two `blobId`. One blobId for  BlobStorageDao parameter and one `blobId` should be computed by hashing payload data. It is separate?